From e7477d99a9fb955ca37bfe3c2318bb64202b6c52 Mon Sep 17 00:00:00 2001
From: Markus Kappeler <markus.kappeler@bytesatwork.ch>
Date: Mon, 15 Mar 2021 13:25:16 +0100
Subject: [PATCH] ARM: dts: stm32mp175c-byteengine/bytedevkit: Set ethernet
 max-speed to 1Gbit

Set correct rx-delay and enable rxctrl-strap-quirk for phy DP83867 needed
for 1Gbit max-speed.

Signed-off-by: Markus Kappeler <markus.kappeler@bytesatwork.ch>
---
 arch/arm/boot/dts/stm32mp157c-bytedevkit.dtsi | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/stm32mp157c-bytedevkit.dtsi b/arch/arm/boot/dts/stm32mp157c-bytedevkit.dtsi
index 8906e882887c..ff310efca916 100644
--- a/arch/arm/boot/dts/stm32mp157c-bytedevkit.dtsi
+++ b/arch/arm/boot/dts/stm32mp157c-bytedevkit.dtsi
@@ -161,7 +161,7 @@
 	phy-mode = "rgmii-id";
 	phy-handle = <&phy0>;
 	st,eth-clk-sel = <1>;
-	max-speed = <100>;
+	max-speed = <1000>;
 	clock-names = "stmmaceth", "mac-clk-tx", "mac-clk-rx", "eth-ck", "syscfg-clk", "ethstp";
 	clocks = <&rcc ETHMAC>, <&rcc ETHTX>, <&rcc ETHRX>, <&rcc ETHCK_K>, <&rcc SYSCFG>, <&rcc ETHSTP>;
 
@@ -171,9 +171,10 @@
 		compatible = "snps,dwmac-mdio";
 		phy0: ethernet-phy@0 {
 			reg = <0>;
-			ti,rx-internal-delay = <DP83867_RGMIIDCTL_2_25_NS>;
+			ti,rx-internal-delay = <DP83867_RGMIIDCTL_3_00_NS>;
 			ti,tx-internal-delay = <DP83867_RGMIIDCTL_2_75_NS>;
 			ti,fifo-depth = <DP83867_PHYCR_FIFO_DEPTH_4_B_NIB>;
+			ti,dp83867-rxctrl-strap-quirk;
 		};
 	};
 };
-- 
2.20.1

